### PR TITLE
Adding support for querying registered/available/busy task executors by SKU ID or any other attribute

### PR DESCRIPTION
**Pull request by sundargates**
### Context

Explain context and other details for this pull request.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable

### CODE REVIEW

File: mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/ResourceCluster.java [hunk 1]
```diff
@@ -54,13 +55,29 @@ public interface ResourceCluster extends ResourceClusterGateway {
      */
     CompletableFuture<Ack> initializeTaskExecutor(TaskExecutorID taskExecutorID, WorkerId workerId);
 
-    CompletableFuture<List<TaskExecutorID>> getRegisteredTaskExecutors();
+    default CompletableFuture<List<TaskExecutorID>> getRegisteredTaskExecutors() {
+        return getRegisteredTaskExecutors(Collections.emptyMap());
+    }
+
+    CompletableFuture<List<TaskExecutorID>> getRegisteredTaskExecutors(Map<String, String> attributes);
```
Comment (Andyz26) [inline mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/ResourceCluster.java:62]: nit: add doc giving attribute sample

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java [hunk 2]
```diff
@@ -759,23 +762,35 @@ public GetActiveJobsRequest(ClusterID clusterID) {
         }
     }
 
+    interface HasAttributes {
```
Comment (Andyz26) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java:765]: nit: this name reads like a bool.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java [hunk 3]
```diff
@@ -975,4 +990,12 @@ private Boolean isJobArtifactCachingEnabled() {
         return true;
 // TODO: fix this ->   return ServiceRegistry.INSTANCE.getPropertiesService().getStringValue("mantis.job.artifact.caching.enabled", "false").equals("true");
     }
+
+    private Predicate<Entry<TaskExecutorID, TaskExecutorState>> filterByAttrs(HasAttributes hasAttributes) {
+        if (hasAttributes.getAttributes().isEmpty()) {
```
Comment (Andyz26) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java:995]: hasAttributes null check?

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java [hunk 4]
```diff
@@ -285,6 +300,19 @@ private Route getResourceOverview(ClusterID clusterID) {
...
     }
 
+    private Route mkTaskExecutorsRoute(
+        ClusterID clusterId,
+        BiFunction<ResourceCluster, GetTaskExecutorsRequest, CompletableFuture<List<TaskExecutorID>>> taskExecutors) {
+        return entity(
+            Jackson.optionalEntityUnmarshaller(GetTaskExecutorsRequest.class),
+            request -> {
+                if (request == null) {
+                    request = new GetTaskExecutorsRequest(ImmutableMap.of());
```
Comment (Andyz26) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java:310]: nit: make this a static instance

**Review by Andyz26 — APPROVED** [review]

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java (hunk 2 above)
Comment (sundargates) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java:765]: Hmm, this is just like a trait, and traits are usually named this way.

**Pull request merged**